app/src/main/java/com/infomaniak/drive/ui/addFiles/CreateFolderFragment.kt

```diff
@@ -38,21 +39,34 @@ import com.infomaniak.drive.ui.MainViewModel
 import com.infomaniak.drive.ui.fileList.fileShare.PermissionsAdapter
 import com.infomaniak.drive.utils.AccountUtils
 import com.infomaniak.drive.utils.showSnackbar
-import com.infomaniak.lib.core.utils.*
 import com.infomaniak.lib.core.utils.ApiErrorCode.Companion.translateError
+import com.infomaniak.lib.core.utils.hideKeyboard
+import com.infomaniak.lib.core.utils.hideProgress
+import com.infomaniak.lib.core.utils.initProgress
+import com.infomaniak.lib.core.utils.showProgress
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.runBlocking
 
 open class CreateFolderFragment : Fragment() {
 
-    protected var binding: FragmentCreateFolderBinding by safeBinding()
+    private var _binding: FragmentCreateFolderBinding? = null
+    protected val binding get() = _binding!! // This property is only valid between onCreateView and onDestroyView
 
     protected val newFolderViewModel: NewFolderViewModel by navGraphViewModels(R.id.newFolderFragment)
     protected val mainViewModel: MainViewModel by activityViewModels()
+
     protected lateinit var adapter: PermissionsAdapter
 
+    private val folderNameTextWatcher by lazy {
+        object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) = Unit
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) = toggleCreateFolderButton()
```

Comment:
Just to be sure do we also check if the binding is null inside `toggleCreateFolderbutton()` and send a Sentry if it's the case? And only execute code if binding is set

Reply:
This looks like over engineering, this should fix this already exceptionnal error case, and we only have 2 sentries about it. I'd rather not putting potentially useless code that we'll forget removing just in case